Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3445,51 +3446,126 @@ def panorama_delete_url_filter_command(url_filter_name: str):
-    if isinstance(rule.get('destination'), dict) and 'member' in rule['destination']:
-        pretty_rule['Destination'] = rule['destination']['member']
-    if isinstance(rule.get('from'), dict) and 'member' in rule['from']:
-        pretty_rule['From'] = rule['from']['member']
-    if isinstance(rule.get('service'), dict) and 'member' in rule['service']:
-        pretty_rule['Service'] = rule['service']['member']
-    if isinstance(rule.get('to'), dict) and 'member' in rule['to']:
-        pretty_rule['To'] = rule['to']['member']
-    if isinstance(rule.get('source'), dict) and 'member' in rule['source']:
-        pretty_rule['Source'] = rule['source']['member']
-    if isinstance(rule.get('tag'), dict) and 'member' in rule['tag']:
-        pretty_rule['Tags'] = rule['tag']['member']
-    if isinstance(rule.get('log-setting'), dict) and '#text' in rule['log-setting']:
-        pretty_rule['LogForwardingProfile'] = rule['log-setting']['#text']
-    if disabled := rule.get('disabled'):
-        pretty_rule['Disabled'] = disabled
-
-    return pretty_rule
+        context_rule['Location'] = rule['@loc']
+        
+    context_rule['SecurityProfileGroup'] = \
+    pretty_rule['Profile Group'] = \
+    context_rule['SecurityProfile'] = \
+    pretty_rule['Profile'] = \
+    None
+    
+    if isinstance(rule.get('profile-setting'), dict):
+        if isinstance(rule['profile-setting'].get('group'), dict):
+            context_rule['SecurityProfileGroup'] = \
+            pretty_rule['Profile Group'] = \
+            rule['profile-setting']['group'].get('member')
+        if isinstance(rule['profile-setting'].get('profiles'), dict):
+            profiles = {
+                k: v.get('member')
+                for k ,v
+                in rule['profile-setting']['profiles'].items()
+                if isinstance(v, dict) and not k.startswith('@')
+            }
+            pretty_rule['Profile'] = '\n'.join(
+                f'{k}: {v}'
+                for k ,v in profiles.items()
+            )
+            context_rule['SecurityProfile'] = profiles
+            
+    if isinstance(rule.get('target'), dict):
+        pretty_rule['Target'] = rule['target']
+        
+        context_rule['Target'] = (
+            f'negate: {rule["target"].get("negate")}\n'
+            f'devices: {xml_get(rule["target"], "devices")}'
+        ).replace('[', '').replace(']', '').replace("'", '')
+    else:
+        context_rule['Target'] = pretty_rule['Target'] = None
+    
+    if option := rule.get('option'):
+        context_rule['Option'] = option
+        pretty_rule['Option'] = 'disable-server-response-inspection: ' +\
+                                str(xml_get(option, 'disable-server-response-inspection'))\
+                                if type(option) is dict \
+                                else None
+    else:
+        context_rule['Option'] = pretty_rule['Option'] = None
+        
+    context_rule['Name'] = pretty_rule['Name'] = rule['@name']
```

Comment:
we dont need to maintain 2 dicts here, as the pretty rules are a subset of the full rules. We should be limiting the human readable by using the chosen headers, not giving a seperate dict

Reply:
Thats what i wanted to do originally but pretty rules is not an exact subgroup
pretty rules mimics the UI and context rules cant break backwards compatibility so their key are different

Reply:
which key are you talking about? Human readable shouldnt be different than the context for the most part

Reply:
for example:
` context_rule['To'] = pretty_rule['Destination Zone'] = rule_get('to')`
there are better examples at the beginning of the function.
Ido stressed in the demo that he wanted the HR to look like the UI so thats why i did it, it also makes more sense because with all the new fields we're adding the old version's fields would be to vague

Reply:
lets use a headerTransform function for this then instead. You can have a dict of from_header_name to to_header_name.

Reply:
the good news is that the keys are the same object in memory so we're not waisting to much space

Reply:
the concern here isnt space, its readability and needless complexity

Reply:
Great tip, will use headerTransform and one dict